x/gamm/pool-models/stableswap/pool.go

```diff
@@ -123,8 +127,14 @@ func (p Pool) getScaledPoolAmts(denoms ...string) ([]sdk.Dec, error) {
 			return []sdk.Dec{}, fmt.Errorf("denom %s does not exist in pool", denom)
 		}
 		scalingFactor := p.GetScalingFactorByLiquidityIndex(liquidityIndex)
+
+		// should not be possible after pool creation checks but we keep this here as a guardrail
+		if int64(scalingFactor) <= 0 {
```

Comment:
nit / non-blocking: at first sight, feels that this might be an overkill / additional complexity given that we already test for this in the constructor, `ValidateBasic`  and `SetStableSwapScalingFactors`

Reply:
I ended up removing this check and the one in `getDescaledPoolAmt` to reduce complexity as all cases should be covered by higher level checks